### ISSUE TITLE
openssl/evp.h header file not found when trying to install mongodb extension

### ISSUE THREAD

**Issue by Partyschaum** [phpbrew/phpbrew#805]
I'm struggling with installing the mongodb extension via `phpbrew --debug ext install mongodb` since it always fails with:

```
In file included from /Users/hauke/.phpbrew/build/php-7.0.12/ext/mongodb/src/contrib/php-ssl.c:31:
/Users/hauke/.phpbrew/build/php-7.0.12/ext/mongodb/src/contrib/php-ssl.h:33:10: fatal error: 'openssl/evp.h' file not found
#include <openssl/evp.h>
         ^
1 error generated.
make: *** [src/contrib/php-ssl.lo] Error 1
```

A gist to the log output is found [here](https://gist.github.com/Partyschaum/83bdb6c18f71b7a029cda71c869b44a6).

I also tried `phpbrew --debug ext install mongodb -- --with-openssl=/usr/local/opt/openssl` which also fails with the above error.

A gist to this log output is found [here](https://gist.github.com/Partyschaum/e3d1ac3835c72d9fb2701449dc32c8b9).

For the corresponding PHP I used `phpbrew --debug install 7.0.12 +default +dbs +openssl=/usr/local/opt/openssl` which seems to work fine. SSL support seems to be working there.

The build log from the PHP compilation is found [here](https://gist.github.com/Partyschaum/0bf2048421a4d7592c6bba2465f7fcc1).

I try all this on an OS X `10.11.6` and phpbrew `1.22.5`.

The SSL is "keg-only" installed via homebrew:

```
openssl: stable 1.0.2j (bottled) [keg-only]
SSL/TLS cryptography library
https://openssl.org/
/usr/local/Cellar/openssl/1.0.1f (1,229 files, 10.6M)
  Poured from bottle on 2014-01-21 at 04:25:33
/usr/local/Cellar/openssl/1.0.1j (1,236 files, 10.6M)
  Poured from bottle on 2014-10-17 at 22:17:12
/usr/local/Cellar/openssl/1.0.1k (1,236 files, 10.7M)
  Poured from bottle on 2015-01-08 at 20:26:52
/usr/local/Cellar/openssl/1.0.2a-1 (1,633 files, 12.3M)
  Poured from bottle on 2015-03-19 at 17:33:54
/usr/local/Cellar/openssl/1.0.2d_1 (1,638 files, 12.1M)
  Poured from bottle on 2015-07-13 at 20:46:23
/usr/local/Cellar/openssl/1.0.2g (1,678 files, 12.0M)
  Poured from bottle on 2016-05-01 at 14:10:22
/usr/local/Cellar/openssl/1.0.2h_1 (1,691 files, 12M)
  Poured from bottle on 2016-06-03 at 20:53:45
/usr/local/Cellar/openssl/1.0.2j (1,695 files, 12M)
  Poured from bottle on 2016-10-10 at 12:33:39
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/openssl.rb
==> Dependencies
Build: makedepend ✘
==> Options
--universal
	Build a universal binary
--without-test
	Skip build-time tests (not recommended)
==> Caveats
A CA file has been bootstrapped using certificates from the SystemRoots
keychain. To add additional certificates (e.g. the certificates added in
the System keychain), place .pem files in
  /usr/local/etc/openssl/certs

and run
  /usr/local/opt/openssl/bin/c_rehash

This formula is keg-only, which means it was not symlinked into /usr/local.

Apple has deprecated use of OpenSSL in favor of its own TLS and crypto libraries

Generally there are no consequences of this for you. If you build your
own software and it requires this formula, you'll need to add to your
build variables:

    LDFLAGS:  -L/usr/local/opt/openssl/lib
    CPPFLAGS: -I/usr/local/opt/openssl/include
    PKG_CONFIG_PATH: /usr/local/opt/openssl/lib/pkgconfig
```

Am I holding it wrong? Any help appreciated!

**Comment by ice6** [comment]
I met same problem, and I believe it is the bug of the mongodb extension tar.

I assumed the following will works,**but it is not**

1. ensure that you installed openssl using brew `brew install openssl`
2. then `phpbrew install 7.0.13 +default +mysql +openssl=/usr/local/opt/openssl`
3. and then, `phpbrew ext install mongodb`

actually, i found that php development env is must complicate on MacOS than on CentOS.
and phpbrew save a lot, but it is still complicate, I think I will use virtual machine on mac to develop php web site.

workaround:
1. download the extension install package, the `tar` file in the console
2. build it(you muck specify the include and lib path in the Makefile)
3. just `phpize -> ./configure -> make -> make install`
4. go to the `var/db` path make a ini just like other ini files

wish you good luck :)

**Comment by arunko350** [comment]
I'm not sure if this thread is still active. I too have spent a day in debugging this issue. I figured out that specifying the openssl dir during the `./configure` step will resolve it. 
So I used below - 
`phpbrew ext install mongodb -- --with-openssl-dir=/usr/local/opt/openssl`

This perfectly worked for me. Many people do the mistake here -  
It is `--with-openssl-dir-` NOT `--with-openssl=`.

Happy Coding!!
